File: FelipezConde/QuestionnaireLanguage/QuestionnaireLanguage/Visitors/Interfaces/ITypeVisitor.cs
using QuestionnaireLanguage.GUI.FormObject;
using Types = AST.Types;

namespace QuestionnaireLanguage.Visitors.Interfaces
{
    public interface ITypeVisitor<T>
    {
        T Visit(Types.StringType question);
        T Visit(Types.IntType conditional);
        T Visit(Types.BoolType conditional);
    }
}

File: Fugazi/src/org/fugazi/ql/ast/type/ITypeVisitor.java
package org.fugazi.ql.ast.type;

public interface ITypeVisitor<T> {
    
    public T visitBoolType(BoolType _boolType);
    public T visitIntType(IntType _intType);
    public T visitStringType(StringType _stringType);
    public T visitUndefinedType(UndefinedType _undefinedType);
}

